xgboost/notebooks/Online Prediction with XGBoost.ipynb

```diff
@@ -0,0 +1,634 @@
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "NAME  DEPLOYMENT_URI                        STATE\r\n",
+      "v1    gs://xgboost-tutorial-notebook-test/  CREATING\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine versions list --model $MODEL_NAME"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "NAME  DEPLOYMENT_URI                        STATE\r\n",
+      "v1    gs://xgboost-tutorial-notebook-test/  READY\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine versions list --model $MODEL_NAME"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Test the model with an online prediction using the data of a person who makes <=50K.\n",
+    "\n",
+    "Note: If you see an error, the model from Part 4 may not be created yet as it takes several minutes for a new model version to be created."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0.09139516949653625]\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine predict --model $MODEL_NAME --version $VERSION_NAME --json-instances less_than_50K.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Test the model with an online prediction using the data of a person who makes >50K."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0.4006653428077698]\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine predict --model $MODEL_NAME --version $VERSION_NAME --json-instances more_than_50K.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Realise how the cells above return floats instead of booleans. Let's deal with that below so the output type of the predictions match those of the test set labels. We'll set the prediction to True if it's greater than 0.5 and to False otherwise."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Python to make online predictions\n",
+    "We'll test the model with the entire test set and print out some of the results.\n",
+    "\n",
+    "Note: If running notebook server on Compute Engine, make sure to [\"allow full access to all Cloud APIs\"](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#changeserviceaccountandscopes)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prediction: False\tLabel: False\n",
```

Comment:
This is a little hard to read. Could you run this through `matplotlib` or `seaborn` to produce a confusion matrix or something similar?

Reply:
hmm how about i show a confusion matrix for both the local and the uploaded models. they would produce the same results, thus showing we uploaded and performed online predictions correctly.

Reply:
@sirtorry That's a great idea.